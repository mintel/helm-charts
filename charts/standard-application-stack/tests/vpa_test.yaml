suite: Test VerticalPodAutoscalers
templates:
  - vpa.yaml
release:
  namespace: test-namespace
tests:
  - it: Doesn't create a VerticalPodAutoscaler
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: false
      statefulset: false
      cronjobsOnly: false
    asserts:
      - hasDocuments:
          count: 0

  - it: Creates a VerticalPodAutoscaler for a Deployment
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      statefulset: false
      cronjobsOnly: false
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: deployment-example-app
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app
      - equal:
          path: spec.updatePolicy.updateMode
          value: "Off"

  - it: Creates a VerticalPodAutoscaler for a StatefulSet
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      statefulset: true
      cronjobsOnly: false
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: statefulset-example-app
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: StatefulSet
      - equal:
          path: spec.targetRef.name
          value: example-app
      - equal:
          path: spec.updatePolicy.updateMode
          value: "Off"

  - it: Only creates VerticalPodAutoscaler for CronJobs if cronjobsOnly is true
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: true
      cronjobs.jobs:
        - name: daily
          concurrencyPolicy: Forbid
          suspend: false
          restartPolicy: Never
          schedule: "0 0 * * *"
          image: ""
          command: [/bin/sh]
          args:
            - "-c"
            - "NO_MIGRATE=1"
            - "/app/docker-entrypoint.sh"
            - "manage"
            - "daily"
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: cronjob-example-app-daily
      - equal:
          path: spec.targetRef.apiVersion
          value: batch/v1
      - equal:
          path: spec.targetRef.kind
          value: CronJob
      - equal:
          path: spec.targetRef.name
          value: example-app-daily
      - equal:
          path: spec.updatePolicy.updateMode
          value: "Off"

  - it: Creates a VerticalPodAutoscaler with autoscaling enabled
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      verticalPodAutoscaler.autoscalingEnabled: true
      statefulset: false
      cronjobsOnly: false
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: deployment-example-app
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app
      - equal:
          path: spec.updatePolicy.updateMode
          value: Auto

  - it: Creates a VerticalPodAutoscaler for aws-es-proxy
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      opensearch.enabled: true
      opensearch.awsEsProxy.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-aws-es-proxy
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-aws-es-proxy
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-aws-es-proxy
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off

  - it: Creates a VerticalPodAutoscaler for celery
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      celery.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-celery
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-celery
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-celery
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off

  - it: Creates a VerticalPodAutoscaler for celery-beat
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      celery.enabled: true
      celeryBeat.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-celery-beat
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-celery-beat
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-celery-beat
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off

  - it: Creates a VerticalPodAutoscaler for celery-exporter
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      celery.enabled: true
      redis.enabled: true
      celery.metrics.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-celery-exporter
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-celery-exporter
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-celery-exporter
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off

  - it: Creates a VerticalPodAutoscaler for mysqlclient
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      mariadb.enabled: true
      mariadb.client.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-mysqlclient
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-mysqlclient
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-mysqlclient
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off

  - it: Creates a VerticalPodAutoscaler for mysqldexporter
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      mariadb.enabled: true
      mariadb.metrics.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-mysqldexporter
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-mysqldexporter
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-mysqldexporter
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off

  - it: Creates a VerticalPodAutoscaler for postgresqlexporter
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      postgresql.enabled: true
      postgresql.metrics.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-postgresqlexporter
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-postgresqlexporter
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-postgresqlexporter
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off

  - it: Creates a VerticalPodAutoscaler for postgresqlclient
    set:
      global.clusterEnv: qa
      verticalPodAutoscaler.enabled: true
      cronjobsOnly: false
      postgresql.enabled: true
      postgresql.metrics.enabled: true
    documentSelector:
      path: metadata.name
      value: deployment-example-app-postgresqlclient
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: VerticalPodAutoscaler
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: deployment-example-app-postgresqlclient
      - equal:
          path: spec.targetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.targetRef.kind
          value: Deployment
      - equal:
          path: spec.targetRef.name
          value: example-app-postgresqlclient
      - equal:
          path: spec.updatePolicy.updateMode
          value: Off
