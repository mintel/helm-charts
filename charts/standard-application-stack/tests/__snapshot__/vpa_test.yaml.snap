Creates a VerticalPodAutoscaler for a Deployment:
  1: |
    apiVersion: autoscaling.k8s.io/v1
    kind: VerticalPodAutoscaler
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: deployment-example-app
      namespace: test-namespace
    spec:
      targetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      updatePolicy:
        updateMode: "Off"
Creates a VerticalPodAutoscaler for a StatefulSet:
  1: |
    apiVersion: autoscaling.k8s.io/v1
    kind: VerticalPodAutoscaler
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: statefulset-example-app
      namespace: test-namespace
    spec:
      targetRef:
        apiVersion: apps/v1
        kind: StatefulSet
        name: example-app
      updatePolicy:
        updateMode: "Off"
Only creates VerticalPodAutoscaler for CronJobs if cronjobsOnly is true:
  1: |
    apiVersion: autoscaling.k8s.io/v1
    kind: VerticalPodAutoscaler
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: daily
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app-daily
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app-daily
      name: cronjob-example-app-daily
      namespace: test-namespace
    spec:
      targetRef:
        apiVersion: batch/v1
        kind: CronJob
        name: example-app-daily
      updatePolicy:
        updateMode: "Off"
